iguana.corecontroller/src/main/java/org/aksw/iguana/cc/tasks/impl/Stresstest.java

```diff
@@ -37,357 +30,307 @@
-		}
-		//wait as long as needed
-		Instant start = Instant.now();
-		exec.shutdown();
-		while(durationInMilliseconds(start, Instant.now()) <= warmupTimeMS) {
-			//clean up RAM
-			for(Worker worker: warmupWorkers) {
-				worker.popQueryResults();
-			}
-			try {
-				TimeUnit.MILLISECONDS.sleep(50);
-			}catch(Exception e) {
-				LOGGER.error("Could not warmup ");
-			}
-		}
-		for(Worker worker : warmupWorkers) {
-			worker.stopSending();
-		}
-		try {
-			exec.awaitTermination(5, TimeUnit.SECONDS);
-
-		} catch (InterruptedException e) {
-			LOGGER.warn("[TaskID: {{}}] Warmup. Could not await Termination of Workers.", taskID);
-		}
-		try {
-			exec.shutdownNow();
-		}catch(Exception e1) {
-			LOGGER.error("Shutdown problems ", e1);
-		}
-		//clear up
-		long queriesExec = 0;
-		for(Worker w : warmupWorkers){
-			queriesExec+=w.getExecutedQueries();
-		}
-		warmupWorkers.clear();
-		LOGGER.info("[TaskID: {{}}] Warmup finished.", taskID);
-		return queriesExec;
-	}
-
-	/**
-	 * Checks if restriction (e.g. timelimit or noOfQueryMixes for each Worker)
-	 * occurs
-	 * 
-	 * @return true if restriction occurs, false otherwise
-	 */
-	protected boolean isFinished() {
-		if (timeLimit !=null) {
-
-			Instant current = Instant.now();
-			double passed_time = timeLimit - durationInMilliseconds(this.startTime, current);
-			return passed_time <= 0D;
-		}
-		else if (noOfQueryMixes != null) {
-
-			// use noOfQueries of SPARQLWorkers (as soon as a worker hit the noOfQueries, it
-			// will stop sending results
-			// UpdateWorker are allowed to execute all their updates
-			boolean endFlag=true;
-			for (Worker worker : workers) {
-				long queriesInMix = 0;
-
-					LOGGER.debug("No of query Mixes: {} , queriesInMix {}", worker.getExecutedQueries(),noOfQueryMixes);
-					//Check for each worker, if the
-					if (worker.hasExecutedNoOfQueryMixes(noOfQueryMixes)) {
-						if(!worker.isTerminated()) {
-							//if the worker was not already terminated, send last results, as tehy will not be sended afterwards
-							sendWorkerResult(worker);
-						}
-						worker.stopSending();
-					}
-					else {
-						endFlag = false;
-					}
-
-			}
-			return endFlag;
-		}
-		LOGGER.error("Neither time limit nor NoOfQueryMixes is set. executing task now");
-		return true;
-	}
-
-	public long getExecutedQueries(){
-		long ret = 0;
-		for(Worker worker: workers){
-			ret += worker.getExecutedQueries();
-		}
-		return ret;
-	}
+    private static final Logger LOGGER = LoggerFactory.getLogger(Stresstest.class);
+
+    private final Map<Object, Object> warmupConfig;
+    private final List<Worker> warmupWorkers = new ArrayList<>();
+    private final List<Map<Object, Object>> workerConfig;
+    protected List<Worker> workers = new LinkedList<>();
+    private Double warmupTimeMS;
+    private Double timeLimit;
+    private Long noOfQueryMixes;
+    private Instant startTime;
+
+
+    public Stresstest(Integer timeLimit, List<Map<Object, Object>> workers) {
```

Comment:
How is workers encoded? Would it make sense to wrap the Map into a type to give it explicit semantics?